applications/rag/frontend/variables.tf

```diff
@@ -129,8 +129,8 @@ variable "support_email" {
 
 variable "domain" {
   type        = string
-  description = "Provide domain for ingress resource and ssl certificate. If it's empty, it will use nip.io wildcard dns"
-  default     = ""
+  description = "Provide domain for ingress resource and ssl certificate."
+  default     = "frontend.example.com"
```

Comment:
I don't think we should default this domain to `example.com`. We should default to blank. It's fine to use in the example `.tfvars` since those are user facing and meant to be edited. I also suspect that defaulting to this might break things or have unexpected issues down the line.

Reply:
Okay we can have that.. This particluar is needed for marketplace defaults..

Reply:
Why does "frontend.example.com" interrupt the terraform execution, but not ""? Wouldn't the managed cert provisioning fail for ""?

Reply:
Can we just turn off IAP with an empty domain? Using `frontend.example.com` isn't going to give you a functioning endpoint.

Reply:
Oh, I misread what Umesh wrote. Yes, I agree that we should not allow the execution to succeed for a fake domain, that's a worse experience than failing early.

Might not be trivial to turn IAP off, but I think the managed cert creation will fail anyway with a specific error if you don't provide a valid domain string. So I'm ok with "".

Reply:
okay lets keep it empty.. and on `""` domain will skip managedcertificate creation.

Reply:
actually, let me include variable validations,, that means if domain is kept empty or has example.com.. then fail the terraform at plan itself.. and fail fast.. so user knows that if IAP is enabled.. then domain is required.